Base/QTCore/Testing/Cxx/qSlicerCoreIOManagerTest1.cxx

```diff
@@ -26,11 +26,62 @@
 #include "qSlicerCoreIOManager.h"
 
 // MRML includes
+#include <vtkMRMLMessageCollection.h>
+#include "vtkMRMLTextNode.h"
 #include "vtkMRMLScene.h"
-#include "vtkMRMLStorableNode.h"
+#include "vtkMRMLStorageNode.h"
 
 #include "vtkMRMLCoreTestingMacros.h"
 
+int TestLongNodeNameSaving(const char* temporaryDirectory)
+{
+  vtkNew<vtkMRMLScene> scene;
+  if (temporaryDirectory)
+  {
+    scene->SetRootDirectory(temporaryDirectory);
+  }
+
+  std::string longNodeName =
+    "Loremipsumdolorsitametconsecteturadipiscingelitseddoeiusmodtemporin"
+    "cididuntutlaboreetdoloremagnaaliquaUtenimadminimveniamquisnostrudex"
+    "ercitationullamcolaborisnisiutaliquipexeacommodoconsequatDuisauteir"
+    "uredolorinreprehenderitinvoluptatevelitessecillumdoloreeufugiatnull"
+    "apariaturExcepteursintoccaecatcupidatatnonproidentsuntinculpaquioff"
+    "iciadeseruntmollitanimidestlaborum";
+  std::string extension = ".txt";
+  std::string longFileName = longNodeName + extension;
+  std::string safeFileName = qSlicerCoreIOManager::forceFileNameValidCharacters(QString::fromStdString(longFileName)).toStdString();
+  safeFileName = qSlicerCoreIOManager::forceFileNameMaxLengthExtension(QString::fromStdString(longFileName), extension.length()).toStdString();
+
+  vtkSmartPointer<vtkMRMLTextNode> textNode = vtkMRMLTextNode::SafeDownCast(
+    scene->AddNewNodeByClass("vtkMRMLTextNode", longNodeName));
+  textNode->SetText(longNodeName);
+  textNode->SetForceCreateStorageNode(true);
+  textNode->AddDefaultStorageNode();
+  vtkMRMLStorageNode* storageNode = textNode->GetStorageNode();
+
+  std::cout << std::endl << "||||||||||||||||||||" << std::endl;
+  std::cout << std::endl << "Testing long file name: " << longFileName << std::endl;
+  storageNode->SetFileName(longFileName.c_str());
+  CHECK_INT(storageNode->WriteData(textNode), 0); // Writing should fail. File name too long.
```

Comment:
I believe this only fails on Windows (and only if long file path support is disabled in the OS, but usually it is).
We could try to make it fail differently on Linux/macOS (for example by injecting some invalid characters into the filepath); or we could just skip these checks on Linux/macOS.

Reply:
Added #ifdef _WIN32 around the section.